lib/chef/provider/package/windows/exe.rb

```diff
@@ -0,0 +1,159 @@
+
+          def installed_packages
+            @installed_packages || begin
+              installed_packages = {}
+              # Computer\HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
+              installed_packages.merge!(extract_installed_packages_from_key(::Win32::Registry::HKEY_LOCAL_MACHINE)) # rescue nil
+              # 64-bit registry view
+              # Computer\HKEY_LOCAL_MACHINE\Software\Wow6464Node\Microsoft\Windows\CurrentVersion\Uninstall
+              installed_packages.merge!(extract_installed_packages_from_key(::Win32::Registry::HKEY_LOCAL_MACHINE, (::Win32::Registry::Constants::KEY_READ | 0x0100))) # rescue nil
+              # 32-bit registry view
+              # Computer\HKEY_LOCAL_MACHINE\Software\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall
+              installed_packages.merge!(extract_installed_packages_from_key(::Win32::Registry::HKEY_LOCAL_MACHINE, (::Win32::Registry::Constants::KEY_READ | 0x0200))) # rescue nil
+              # Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall
+              installed_packages.merge!(extract_installed_packages_from_key(::Win32::Registry::HKEY_CURRENT_USER)) # rescue nil
+              installed_packages
+            end
+          end
+
+          # http://unattended.sourceforge.net/installers.php
+          def unattended_installation_flags
+            case @installer_type
+            when :installshield
+              '/s /sms'
+            when :nsis
+              '/S /NCRC'
+            when :inno
+              # "/sp- /silent /norestart"
+              '/verysilent /norestart'
+            when :wise
+              '/s'
+            end
+          end
+
+          def extract_installed_packages_from_key(hkey = ::Win32::Registry::HKEY_LOCAL_MACHINE, desired = ::Win32::Registry::Constants::KEY_READ)
+            uninstall_subkey = 'Software\Microsoft\Windows\CurrentVersion\Uninstall'
+            packages = {}
+            begin
+              ::Win32::Registry.open(hkey, uninstall_subkey, desired) do |reg|
+                reg.each_key do |key, _wtime|
+                  begin
+                    k = reg.open(key, desired)
+                    display_name = begin
+                                     k['DisplayName']
+                                   rescue
+                                     nil
+                                   end
+                    version = begin
+                                k['DisplayVersion']
+                              rescue
+                                'NO VERSION'
+                              end
+                    uninstall_string = begin
+                                         k['UninstallString']
+                                       rescue
+                                         nil
+                                       end
+                    if display_name
+                      packages[display_name] = { name: display_name,
```

Comment:
note that the usage of display_name as the key here prevents handle two versions of a package with the same name. this is an existing bug, but consider fixing it if doing so would break something.

https://github.com/chef-cookbooks/windows/issues/192